utility/utils.py

```diff
@@ -662,3 +665,55 @@ def download_file(url, filename):
         r = requests.get(url)
         f.write(r.content)
     assert r.ok
+
+
+def destroy_cluster(cluster_path):
+    """
+    Destroy existing cluster resources in AWS.
+
+    Args:
+        cluster_path (str): filepath to cluster directory to be destroyed
+
+    Returns:
+        TestStatus: enum for status of cluster deletion
+
+    """
+    destroy_cmd = (
+        f"./openshift-install destroy cluster "
+        f"--dir {cluster_path} "
+        f"--log-level debug"
+    )
+    installer_filename = "openshift-install"
+    tarball = f"{installer_filename}.tar.gz"
+
+    try:
+        cluster_path = os.path.normpath(cluster_path)
+
+        # Retrieve cluster name and aws region from metadata
+        metadata_file = os.path.join(cluster_path, "metadata.json")
+        with open(metadata_file) as f:
+            metadata = json.loads(f.read())
+        cluster_name = metadata.get("clusterName")
+        region_name = metadata.get("aws").get("region")
+
+        # Execute destroy cluster using OpenShift installer
+        log.info(f"Destroying cluster defined in {cluster_path}")
+        run_cmd(destroy_cmd)
+
```

Comment:
can we have check whether openshift installer is present or not. If not present we need to download it again. The reason why i am asking is if the user clones the repo and use the destroy functionality, it will fail.

Also we can we make download installer as separate lib in utility with version as parameter like
def download_installer(installer_version):

Reply:
Downloading the installer should definitely be it's own utility, I agree. I can break that out and then call it in the event that the installer doesn't exist. This will be useful since we clean up the installer after destroying the cluster as well.